### PR TITLE
Fix temperature cards not syncing

### LINKED ISSUE #18455: Thermostat card doesn't reset values when an exception is thrown
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

If an exception is thrown when setting a new temperature, the temperature card does not return to its original values.

https://github.com/home-assistant/frontend/assets/37427166/2abf5637-edbe-47ab-880b-b578d2dcf246

### Describe the behavior you expected

When an exception occurs while trying to set the temperature, the user interface should revert to the original value(s).

### Steps to reproduce the issue

1. Open Home Assistant on a web browser
2. Add a thermostat card to the user interface
3. Act on the thermostat that invokes an exception/causes the action to fail
4. **Observe** that the values on the UI are what the user provided, despite the actual values being different.

### What version of Home Assistant Core has the issue?

Core versions:
- 2023.12.0.dev0
- 2023.10.2

Frontend versions:
- 20231027.0
- 20231005.0

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Google Chrome 118.0.5993.70 (Official Build) (64-bit) 

### Which operating system are you using to run this browser?

Arch Linux 6.5.6-arch2-1

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

This may be a regression bug. See: https://github.com/home-assistant/frontend/issues/15483, https://github.com/home-assistant/frontend/pull/15497

### PR DESCRIPTION
**Pull request by dbolger**
## Proposed change

Ensure the thermostat tile card, the thermostat card, and the thermostat dialog card keep their original values when an exception is thrown.

Demo:

https://github.com/home-assistant/frontend/assets/37427166/1236031f-f1da-460a-8b13-ce9bd5e5c47e

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
type: thermostat
entity: climate.nest_thermostat
```
```yaml
type: tile
entity: climate.nest_thermostat
tap_action:
  action: call-service
  service: ''
vertical: false
features:
  - type: climate-hvac-modes
    hvac_modes:
      - 'off'
      - heat
      - cool
      - heat_cool
  - type: target-temperature
```

## Additional information

- This PR fixes or closes issue: fixes #18455
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 1]
```diff
@@ -487,9 +489,11 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
     }
 
     data.entity_id = this._config!.entity;
-
-    await this.hass!.callService("climate", service, data);
-
+    try {
+      await this.hass!.callService("climate", service, data);
+    } catch (err: any) {
+      this._error = err.message;
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-thermostat-card.ts:495]: This is not used in render?

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-thermostat-card.ts:495]: Also this should be lokalized :-)

Comment (dbolger) [reply src/panels/lovelace/cards/hui-thermostat-card.ts:495]: Correct, it's not used in render. But when an exception is thrown and not caught, this method doesn't finish executing and as a result, doesn't revert the UI back to its original state.

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-thermostat-card.ts:495]: you can catch the error and swallow? No need to set it as a property?

Comment (dbolger) [reply src/panels/lovelace/cards/hui-thermostat-card.ts:495]: I was following the convention I saw in other cards. Maybe setting the error isn't necessary but I don't think swallowing an exception is a good practice.

**Review by silamon — COMMENTED** [review]
The target temperature tile feature (or card feature in newer releases) does also support water heater entities. For the linting to pass, you may need to add the attributes "target_temp_low", and "target_temp_high" to the WaterHeaterEntity class.

**Pull request closed without merging**